core/src/main/scala/kafka/controller/ControllerContext.scala

```diff
@@ -80,14 +86,49 @@ class ControllerContext {
     replicaStates.clear()
   }
 
+  def partitionReplicaAssignment(topicPartition: TopicPartition): Seq[Int] = {
+    partitionAssignments.getOrElse(topicPartition.topic, mutable.Map.empty)
+      .get(topicPartition.partition) match {
+        case Some(partitionAssignment) => partitionAssignment.replicas
+        case None => Seq.empty
+      }
+  }
+
+  def partitionFullReplicaAssignment(topicPartition: TopicPartition): PartitionReplicaAssignment = {
```

Comment:
ideally, I would call this method `partitionReplicaAssignment` and rename the other to `partitionReplicas`. I didn't go ahead with the change because the diff is already big and I wanted to make it easier for reviewers.

ditto for the update methods below - `updatePartitionReplicaAssignment` and `updatePartitionReplicas`

We could change it after initial reviews